src/WebJobs.Script.WebHost/Management/WebFunctionsManager.cs

```diff
@@ -146,7 +146,8 @@ await functionMetadata
             }
 
             // we need to sync triggers if config changed, or the files changed
-            await _functionsSyncManager.TrySyncTriggersAsync();
+            // we do not want to wait until TrySyncTriggersAsync is finished, there can be several function creations at once
```

Comment:
This doesn't seem right. Why does this fix the issue? With this, we'd be doing things inconsistently - all the other places (including other places in this file) we await.

Reply:
I provided some feedback on this in person. The ideal approach is to await, but suspend the restart until this call completes.

Reply:
If the underlying issue is disposal of scopes, then this doesn't seem specific to SyncTriggers at all. Any other async operation invoked by a service in the DI container can run into similar issues?

Reply:
The fix is for a CRI, the customer deploys few functions files inside ARM template. As we allow only one SyncTriggers call in once (~3 secs) the deployment take to much time and ends up with the dryloc container disposed exception. We do not rely on  SyncTriggers result in the function creation flow so the fire-end-forget SyncTrigger calls fix this particular issue.

>Any other async operation invoked by a service in the DI container can run into similar issues?
Yes, if we restarting the host and the remaining operation takes > 5 secs.

>The ideal approach is to await, but suspend the restart until this call completes.
We use file watchers for the restart and suspending restart will be a bit tricky. Do you have an idea how to achive this?

Reply:
The above is not entirely accurate. In this particular case, the only reason the scope is being disposed is because this API logic is specifically triggering a restart. That would not happen in other async operations and is not an issue there.

Reply:
@alrod the logic is all driven by the file monitoring service. If you enhance the service to suspend notifications, the restart wouldn't be triggered by the operations taking place during the API call. You'd be required to explicitly restart the host at the end, though.

Reply:
As discussed with @fabiocav offline we are OK with a practical fix now, but ideally the proper fix is adding new methods to `FileMonitorService` which will insrease/descrease count of threads requested for the host restart suspending and restart the host only if the count = 0

Reply:
Since we're not doing a release this sprint, we have time to fix this properly right? Also, whatever we do, we'd want to do it consistently as I mentioned above for Delete as well, where we're also doing a sync.

Reply:
Send the PR with "proper" fix:
https://github.com/Azure/azure-functions-host/pull/6799/files